libkbfs/journal_md_ops.go

```diff
@@ -183,6 +217,12 @@ func (j journalMDOps) getRangeFromJournal(
 	irmds := make([]ImmutableRootMetadata, 0, len(ibrmds))
 
 	for _, ibrmd := range ibrmds {
+		if bid == LocalSquashBranchID && ibrmd.BID() != bid {
```

Comment:
It seems that you'll run into the check on line 201 first, though, so you may want to do this filtering before then.